src/Cohesity.Powershell/Scripts/StorageDomain/Set-CohesityStorageDomain.ps1

```diff
@@ -1,19 +1,57 @@
+        Update storage domain (view box) with user specified deduplication, Compression disabled. Based on enable/disable state of compression parameter, compression policy will be decided.
+        .EXAMPLE
+        Get-CohesityStorageDomain | Set-CohesityStorageDomain -Name <DomainName> -PhysicalQuota 20
+        Get the list of storage domain through pipeline and fetch the specified storage domain domain. Update the physical quota for the specified storage domain (view box).
+    #>
     [CmdletBinding()]
     Param(
         [Parameter(Mandatory=$False)]
         [string]$Name = $null,
+        # New domain name to be update
+        [Parameter(Mandatory=$False, ParameterSetName = 'UpdateField')]
+        [string]$NewDomainName = $null,
+        # If deduplication is enabled, the Cohesity Cluster eliminates duplicate blocks of repeating data stored on the Cluster,
+        # thus reducing the amount of storage space needed to store data.
+        [Parameter(Mandatory = $False, ParameterSetName = 'UpdateField')][ValidateSet("true", "false")]
+        [String]$Deduplication = $true,
+        # Specifies if deduplication should occur inline (as the data is being written). This field is only relevant if deduplication is enabled.
+        # If on, deduplication occurs as the Cluster saves blocks to the Partition. If off, deduplication occurs after the Cluster writes data to the Partition.
+        [Parameter(Mandatory = $False, ParameterSetName = 'UpdateField')][ValidateSet("true", "false")]
+        [String]$InlineDeduplication,
+        # Determines whether the compression policy should be ‘kCompressionNone’ (disabled case) or ‘kCompressionLow’ (enabled case)
+        # ‘kCompressionNone’ indicates that data is not compressed. ‘kCompressionLow’ indicates that data is compressed.
+        [Parameter(Mandatory = $False, ParameterSetName = 'UpdateField')][ValidateSet("true", "false")]
+        [String]$Compression,
+        # Specifies if compression should occur inline (as the data is being written). This field is only relevant if compression is enabled.
+        # If on, compression occurs as the Cluster saves blocks to the Partition. If off, compression occurs after the Cluster writes data to the Partition.
+        [Parameter(Mandatory = $False, ParameterSetName = 'UpdateField')][ValidateSet("true", "false")]
+        [String]$InlineCompression,
+        # Specifies an optional quota limit on the usage allowed for this resource. This limit is specified in GiB.
+        # If no value is specified,there is no limit.
+        [Parameter(Mandatory = $False, ParameterSetName = 'UpdateField')]
+        [Int]$PhysicalQuota = $null,
+        # List of storage domain (view box), returned from 'Get-CohesityStorageDomain' commandlet
         [Parameter(ValueFromPipeline=$True, DontShow=$True)]
 		[object[]]$StorageDomain = $null
```

Comment:
It could be an array or single object.. does this parse for both ?

Reply:
Yes, it will parse both. 
Note: As this is the pipeline output, we will get only one object at a time from the pipeline